threads.go

```diff
@@ -1,153 +1,132 @@
 package bt
 
 import (
-	"bytes"
+	"fmt"
 	"os"
-	"regexp"
-	"strconv"
+	"strings"
 )
 
-var (
-	threadRegex      = regexp.MustCompile(`^(.*):$`)
-	fnRegex          = regexp.MustCompile(`^(.*)\.([^(]+)`)
-	createdByFnRegex = regexp.MustCompile(`^created by (.*)\.([^ ]+) in goroutine [0-9]+$`)
-	srcRegex         = regexp.MustCompile(`^\s*(.*):(\d+)`)
-)
-
-type thread struct {
-	id     int
-	name   string
-	frames []frame
+type Thread struct {
+	Name   string       `json:"name"`
+	Stacks []StackFrame `json:"stack"`
```

Comment:
there is also a "fault" boolean variable associated with a thread

![image](https://github.com/user-attachments/assets/4e670934-5a49-4e71-b795-9b6bc2d35cb2)